wrapper/src/main/java/software/amazon/jdbc/plugin/customendpoint/CustomEndpointMonitorImpl.java

```diff
@@ -0,0 +1,238 @@
+   */
+  @Override
+  public void run() {
+    LOGGER.fine(
+        Messages.get(
+            "CustomEndpointMonitorImpl.startingMonitor",
+            new Object[] { this.customEndpointHostSpec.getHost() }));
+
+    try {
+      while (!this.stop.get() && !Thread.currentThread().isInterrupted()) {
+        long start = System.nanoTime();
+
+        final Filter customEndpointFilter =
+            Filter.builder().name("db-cluster-endpoint-type").values("custom").build();
+        final DescribeDbClusterEndpointsResponse endpointsResponse =
+            this.rdsClient.describeDBClusterEndpoints(
+                (builder) ->
+                    builder.dbClusterEndpointIdentifier(this.endpointIdentifier).filters(customEndpointFilter));
+
+        List<DBClusterEndpoint> endpoints = endpointsResponse.dbClusterEndpoints();
+        if (endpoints.size() != 1) {
+          List<String> endpointURLs = endpoints.stream().map(DBClusterEndpoint::endpoint).collect(Collectors.toList());
+          LOGGER.warning(
+              Messages.get("CustomEndpointMonitorImpl.unexpectedNumberOfEndpoints",
+                  new Object[] {
+                      this.endpointIdentifier,
+                      this.region.id(),
+                      endpoints.size(),
+                      endpointURLs
+                  }
+              ));
+
+          TimeUnit.NANOSECONDS.sleep(this.refreshRateNano);
+          continue;
+        }
+
+        CustomEndpointInfo endpointInfo = CustomEndpointInfo.fromDBClusterEndpoint(endpoints.get(0));
+        CustomEndpointInfo cachedEndpointInfo = customEndpointInfoCache.get(this.customEndpointHostSpec.getHost());
+        if (cachedEndpointInfo != null && cachedEndpointInfo.equals(endpointInfo)) {
+          long elapsedTime = System.nanoTime() - start;
+          long sleepDuration = Math.min(0, this.refreshRateNano - elapsedTime);
+          TimeUnit.NANOSECONDS.sleep(sleepDuration);
+          continue;
+        }
+
+        LOGGER.fine(
+            Messages.get(
+                "CustomEndpointMonitorImpl.detectedChangeInCustomEndpointInfo",
+                new Object[]{ this.customEndpointHostSpec.getHost(), endpointInfo }));
+
+        // The custom endpoint info has changed, so we need to update the plugin service info cache.
+        customEndpointInfoCache.put(
+            this.customEndpointHostSpec.getHost(), endpointInfo, CUSTOM_ENDPOINT_INFO_EXPIRATION_NANO);
+        this.pluginService.setInfo(this.customEndpointHostSpec.getHost(), endpointInfo, true);
+        this.infoChangedCounter.inc();
+
+        long elapsedTime = System.nanoTime() - start;
+        long sleepDuration = Math.min(0, this.refreshRateNano - elapsedTime);
+        TimeUnit.NANOSECONDS.sleep(sleepDuration);
+      }
+    } catch (InterruptedException e) {
+      LOGGER.info(Messages.get("CustomEndpointMonitorImpl.interrupted", new Object[]{ this.customEndpointHostSpec }));
+      Thread.currentThread().interrupt();
```

Comment:
This catch block is outside of the while loop so it will just return after this line even without an explicit return statement. If I add a return statement my IDE highlights it as being unnecessary.